src/microbit/model/display.py

```diff
@@ -0,0 +1,208 @@
+from .image import Image
+from .. import shim
+
+
+class Display:
+    # The implementation based off of https://github.com/bbcmicrobit/micropython/blob/master/docs/display.rst.
+    def __init__(self):
+        self.__image = Image()
+        self.__on = True
+
+    def scroll(self, value, delay=150, wait=True, loop=False, monospace=False):
+        if not wait:
+            thread = threading.Thread(
+                target=self.scroll, args=(value, delay, True, loop, monospace)
+            )
+            thread.start()
+            return
+        while True:
+            if isinstance(value, (str, int, float)):
+                value = str(value)
+            else:
+                raise TypeError(f"can't convert {type(value)} object to str implicitly")
+            letters = []
+            for c in value:
+                if monospace:
+                    if c == " ":
+                        letters.append(Image(6, CONSTANTS.LED_HEIGHT))
+                    else:
+                        letters.append(Display.__get_image_from_char(c))
+                        letters.append(Image(1, CONSTANTS.LED_HEIGHT))
+                else:
+                    if c == " ":
+                        letters.append(Image(3, CONSTANTS.LED_HEIGHT))
+                    else:
+                        letters.append(
+                            Display.__strip_image(Display.__get_image_from_char(c))
+                        )
+                        letters.append(Image(1, CONSTANTS.LED_HEIGHT))
+            appended_image = Display.__create_scroll_image(letters)
+            # Show the scrolled image one square at a time.
+            for x in range(appended_image.width() - CONSTANTS.LED_WIDTH + 1):
+                self.__image.blit(
+                    appended_image, x, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
+                )
+                time.sleep(delay / 1000)
+            if not loop:
+                break
+
+    def show(self, value, delay=400, wait=True, loop=False, clear=False):
+        if not wait:
+            thread = threading.Thread(
+                target=self.show, args=(value, delay, True, loop, clear)
+            )
+            thread.start()
+            return
+        while True:
+            if isinstance(value, Image):
+                self.__image = value.crop(
+                    0, 0, CONSTANTS.LED_WIDTH, CONSTANTS.LED_HEIGHT
+                )
+            elif isinstance(value, (str, int, float)):
+                if isinstance(value, str):
+                    chars = list(value)
+                else:
+                    chars = list(str(value))
+
+                for c in chars:
+                    self.__image = Display.__get_image_from_char(c)
+                    time.sleep(delay / 1000)
+            else:
+                # Check if iterable
+                try:
+                    _ = iter(value)
+                except TypeError as e:
+                    raise e

```

Comment:
I would probably do the prep for displaying outside of the loop and keep only the display stuff in the loop.
It's probably easier to understand and prevents logic duplication.